src/array.rs

```diff
@@ -57,6 +72,12 @@ impl<T: RuntimeType> Array<T> {
         self.clear();
         &mut self.data as *mut _ as *mut _
     }
+
+    pub unsafe fn into_abi(self) -> (*mut T::Abi, u32) {
```

Comment:
The std lib usually marks functions like this as safe since it's only unsafe to _use_ the raw pointer not to produce it. I'm fine with keeping a more conservative approach to this, but then we need to document its safety properties.